internal/config/incremental_sync.go

```diff
@@ -0,0 +1,259 @@
+package config
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/icinga/icinga-go-library/database"
+	"github.com/icinga/icinga-go-library/types"
+	"github.com/jmoiron/sqlx"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"time"
+)
+
+// IncrementalConfigurable specifies Getter methods required for types supporting incremental configuration loading.
+type IncrementalConfigurable[PK comparable] interface {
+	zapcore.ObjectMarshaler
+
+	// GetPrimaryKey returns the primary key value.
+	GetPrimaryKey() PK
+
+	// GetChangedAt returns the changed_at value.
+	GetChangedAt() types.UnixMilli
+
+	// IsDeleted returns if this entry was marked as deleted.
+	IsDeleted() bool
+}
+
+// IncrementalConfigurableInitAndValidatable defines a single method for new and updated elements to allow both
+// initialization and validation, to be used within incrementalFetch.
+type IncrementalConfigurableInitAndValidatable interface {
+	// IncrementalInitAndValidate allows both to initialize and validates with an optional error.
+	//
+	// If an error is returned, the incrementalFetch function aborts the element in question.
+	IncrementalInitAndValidate() error
+}
+
+// incrementalFetch queries all recently changed elements of BaseT and stores them in changeConfigSetField.
+//
+// The RuntimeConfig.configChangeTimestamps map contains the last known timestamp for each BaseT table. Only those
+// elements where the changed_at SQL column is greater than the stored timestamp will be fetched and stored in the
+// temporary RuntimeConfig.configChange ConfigSet. Later on, incrementalApplyPending merges it into the main ConfigSet.
+func incrementalFetch[
+	BaseT any,
+	PK comparable,
+	T interface {
+		*BaseT
+		IncrementalConfigurable[PK]
+	},
+](ctx context.Context, tx *sqlx.Tx, r *RuntimeConfig, changeConfigSetField *map[PK]T) error {
+	startTime := time.Now()
+
+	var typePtr T
+
+	tableName := database.TableName(typePtr)
+	changedAt, hasChangedAt := r.configChangeTimestamps[tableName]
+
+	stmtLogger := r.logger.With(zap.String("table", tableName))
+
+	stmt := r.db.BuildSelectStmt(typePtr, typePtr)
+	stmtArgs := []any{}
```

Comment:
```suggestion
	var stmtArgs []any
```

No need to initialize the slice here, `nil` works just fine and doesn't allocate[^1] something that's thrown away in most cases anyways and is not necessary in the remaining case.

[^1]: Maybe the compiler is smart enough to optimize it away, but I don't see a downside of just using `nil` here and that wouldn't even need a smart compiler.